### ISSUE TITLE
Redirect works wrong for uri without protocol

### ISSUE THREAD

**Issue by demon101** [grails/grails-core#10497]
### Task List

- [X] Steps to reproduce provided
- [ ] Example that reproduces the problem uploaded to Github
- [X] Full description of the issue provided (see below)

### Steps to Reproduce

1. redirect(uri: "//google.com") or redirect(url: "//google.com") 

### Expected Behaviour

redirect to current-app-protocol://google.com or //google.com

### Actual Behaviour

 current-app-address//google.com  . For example http://my-app.com//google.com

### Environment Information

- **Grails Version:** 3.1.15

**Comment by demon101** [comment]
@graemerocher  I think the issue -- not improvement. 
Now https used more often (thx google), and urls with related protocol become more usual.

**Comment by graemerocher** [comment]
It is an improvement because you are arguing the difference between `redirect(url: "//google.com")` and `redirect(url: "https://google.com")`. If HTTPS is generally accepted as better there is no need to even factor in the protocol, just always redirect to HTTPS

Besides it could well be a breaking change because `redirect(uri: "//google.com")` is not a like complete URL and is in fact a URI so does it means `https://foo.com//google.com` or `https://google.com`? It is ambiguous and although you could argue that the double slash is the differentiator it could still break existing applications if we change it.

**Comment by demon101** [comment]
redirect(uri: "//google.com") should redirect to currentRequestProtocol+"//google.com"

**Comment by graemerocher** [comment]
Yes that is the opinion if you have stated in this issue, but that doesn't change the fact it is a potentially breaking change.

**Comment by demon101** [comment]
@graemerocher  agree =(

**Comment by owenrubel** [comment]
Requestforwarder had this functionality removed in 3.1 I believe

**Comment by owenrubel** [comment]
Hmmm actually I am correct in my research.
URI's are handled by forward which handles it internally 'forwarding' to an internal class/action. This is because we already have the token for authorization against the endpoint and all request data that we should need so the request should not need to be reissued.

URL's are redirected as they need to check against a new endpoint and may need to reissue a token and get new request data.

Thus the URI should be associated with the RequestForwarder.

This functionality makes no sense as the request/response does not need to be reissued, CORS does not need to be checked again, etc. A URI is a 'forward' to an internal controller/action

**Comment by owenrubel** [comment]
As **Spring (the makers of Spring Boot)** put it:
"forward happens entirely on a server side..."
Thus a **URI, is most definitely a forward... not a redirect**.

This is bad functionality.

**Comment by davydotcom** [comment]
would this be something that makes sense in grails 4.1.0 since its a potentially breaking change
